### ISSUE TITLE
what is the meaning of dtype parameter in get_symbol() function?

### ISSUE THREAD

**Issue by universewill** [apache/incubator-mxnet#7703]
Here is the code. I do not understand what this code does.
I  read an image from opencv python, an it prints uint8.
Do i have to specify the dtype parameter when calling get_symbol function?

'''
def get_symbol(num_classes=1000, dtype='float32', **kwargs):
    data = mx.sym.Variable(name="data")
    if dtype == 'float32':
        data = mx.sym.identity(data=data, name='id')
    else:
        if dtype == 'float16':
            data = mx.sym.Cast(data=data, dtype=np.float16)
'''

**Comment by Godricly** [comment]
You don't have to since the float32 is the default setup. In most cases, we use float type in training. And the image is normalized into range [0,1]. float32 here means typical single precision float type, while float16 represents half precision float point. It requires less memory on GPU and runs faster for some Nvidia GPUs with half precision units. However, due to the loss of precision, the performance could be worse than single precision one.

**Comment by universewill** [comment]
ok, thank you @Godricly

**Comment by universewill** [comment]
Shoud i normalize the image before put it as data input? Or the normalize process will automatically down by mxnet? @Godricly 
I see the code in preprocessing.py in inception-v3  pretraned model tar file (The pretrainded model is in mxnet model zoo). Should i do the same before using the inception-v3  pretraned model?

'''
    # sub mean
    normed_img = sample - 128.
    normed_img /= 128.
'''
